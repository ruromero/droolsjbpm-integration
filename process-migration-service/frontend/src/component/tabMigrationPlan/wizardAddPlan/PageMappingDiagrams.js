import React from "react";
import PropTypes from "prop-types";
import { Button } from "patternfly-react";

import PageMappingDiagramSvgPan from "./PageMappingDiagramSvgPan";
import { ControlLabel } from "patternfly-react/dist/js/components/Form";
import Col from "patternfly-react/dist/js/components/Grid/Col";

export default class PageMappingDiagrams extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showSourceDiagram: false,
      showTargetDiagram: false
    };
  }

  toggleShowSourceDiagram = () => {
    this.setState({
      showSourceDiagram: !this.state.showSourceDiagram
    });
  };

  toggleShowTargetDiagram = () => {
    this.setState({
      showTargetDiagram: !this.state.showTargetDiagram
    });
  };

  render() {
    return (
      <React.Fragment>
        <Col sm={4} className="text-right">
          <Button onClick={this.toggleShowSourceDiagram}>
            {this.state.showSourceDiagram ? "Hide" : "Show"} Source Diagram
          </Button>
        </Col>
        <Col sm={4} className="text-right">
          <Button onClick={this.toggleShowTargetDiagram}>
            {this.state.showTargetDiagram ? "Hide" : "Show"} Target Diagram
          </Button>
        </Col>
        {this.state.showSourceDiagram && (
          <div className="col-sm-12">
            <Col componentClass={ControlLabel}>Source diagram</Col>
            <PageMappingDiagramSvgPan
              svg={this.props.sourceProcess.svgFile}
              previous={this.props.sourcePreviousNode}
              current={this.props.sourceCurrentNode}
            />
          </div>
        )}
        {this.state.showTargetDiagram && (
          <div className="col-sm-12">
            <Col componentClass={ControlLabel}>Target diagram</Col>
            <PageMappingDiagramSvgPan
              svg={this.props.targetProcess.svgFile}
              previous={this.props.targetPreviousNode}
              current={this.props.targetCurrentNode}
            />
          </div>
        )}
      </React.Fragment>
    );
  }
}
PageMappingDiagrams.defaultProps = {
  sourcePreviousNode: "",
  sourceCurrentNode: "",
  targetPreviousNode: "",
  targetCurrentNode: ""
};

PageMappingDiagrams.propTypes = {
  sourceProcess: PropTypes.object.isRequired,
  targetProcess: PropTypes.object.isRequired,
  sourcePreviousNode: PropTypes.string,
  sourceCurrentNode: PropTypes.string,
  targetPreviousNode: PropTypes.string,
  targetCurrentNode: PropTypes.string
};
