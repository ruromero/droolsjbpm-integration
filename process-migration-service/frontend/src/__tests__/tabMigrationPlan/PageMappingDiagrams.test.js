import renderer from "react-test-renderer";
import React from "react";
import PageMappingDiagrams from "../../component/tabMigrationPlan/wizardAddPlan/PageMappingDiagrams";
import evaluationDef from "../../../mock_data/process_definition_evaluation.json";
import mortgageDef from "../../../mock_data/process_definition_mortgage.json";

test("PageMappingDiagrams renders correctly using snapshot", () => {
  const tree = renderer
    .create(
      <PageMappingDiagrams
        sourceProcess={evaluationDef}
        targetProcess={mortgageDef}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
