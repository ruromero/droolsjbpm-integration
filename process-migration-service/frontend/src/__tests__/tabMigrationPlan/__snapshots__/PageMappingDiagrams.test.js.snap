// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMappingDiagrams renders correctly using snapshot 1`] = `
Array [
  <div
    className="text-right col-sm-4"
  >
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Show
       Source Diagram
    </button>
  </div>,
  <div
    className="text-right col-sm-4"
  >
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Show
       Target Diagram
    </button>
  </div>,
]
`;
